lib/rule-tester/rule-tester.js

```diff
@@ -644,13 +644,22 @@ class RuleTester {
             assert.ok(item.errors || item.errors === 0,
                 `Did not specify errors for an invalid test of ${ruleName}`);
 
+            if (Array.isArray(item.errors) && item.errors.length === 0) {
+                assert.fail("errors array should have at least one element otherwise its a valid case");
+            }
+
             const ruleHasMetaMessages = hasOwnProperty(rule, "meta") && hasOwnProperty(rule.meta, "messages");
             const friendlyIDList = ruleHasMetaMessages ? `[${Object.keys(rule.meta.messages).map(key => `'${key}'`).join(", ")}]` : null;
 
             const result = runRuleForItem(item);
             const messages = result.messages;
 
             if (typeof item.errors === "number") {
+
+                if (item.errors === 0) {
+                    assert.fail("errors of type `number` should not have value `0`");
```

Comment:
This error message explains what but not why. Can this be the same error message as above, or is there maybe a better way to explain why this error message is being reported?

Reply:
I changed as mentioned above.
 for `why` , is something like `Invalid cases must have value greater than 0 as having 0 can result in false positive` sounds better  ?

Reply:
Could we just report the same message as above? `"Invalid cases must have at least one error"`?

Reply:
umm,
atleast one error might not specify non-zero number, right ? cause here it should be > 0 if its of type `number` .

WDYT ?

Reply:
I see. Can we explain why a false positive is possible? My concern here is that this error message doesn't give a lot of context around what the actual problem is, but instead describes the implementation detail.

Reply:
What I think is, 

for error type `number` ,

if its `0`  (`items.errors = 0`), and for valid case `messages.length` is also `0` so we are doing `assert.strictEqual(message.lenght, items.errors)` which will be `true`

and for `Array<object>` type, 
it iterates over each, so if the case is `valid` , there won't be any iteration as  `item.errors.length` is `0` so there won't be any assertion (check) so it will be false positive. 
 
Now, we need to compress this to a short message for the explanation 😄.
cause we cant have these much long detail right ?
any suggestion for this would be awesome.

Reply:
cc @kaicataldo thoughts !?

Reply:
Sorry, will try to get to this next week!

Reply:
cool 👍